docs/source/jit_type_system_1.rst

```diff
@@ -0,0 +1,628 @@
+.. contents::
+    :local:
+    :depth: 2
+
+
+.. testsetup::
+
+    # These are hidden from the docs, but these are necessary for `doctest`
+    # since the `inspect` module doesn't play nicely with the execution
+    # environment for `doctest`
+    import torch
+
+    original_script = torch.jit.script
+    def script_wrapper(obj, *args, **kwargs):
+        obj.__module__ = 'FakeMod'
+        return original_script(obj, *args, **kwargs)
+    torch.jit.script = script_wrapper
+
+    original_trace = torch.jit.trace
+    def trace_wrapper(obj, *args, **kwargs):
+        obj.__module__ = 'FakeMod'
+        return original_trace(obj, *args, **kwargs)
+    torch.jit.trace = trace_wrapper
+
+
+.. _type_system:
+
+
+Type System (I): TorchScript Types
+==================================
+
+This section summarizes TorchScript type system. The information is based on TorchScript language reference
+and reading the implementation.
+
+Terminology
```

Comment:
Since these terminologies are used throughout the language specification, maybe better to move them into a separate section in the beginning.